lib/ClusterShell/CLI/Clush.py

```diff
@@ -680,8 +685,16 @@ def run_command(task, cmd, ns, timeout, display, remote, trytree):
                         remote=remote, tree=trytree)
     if ns is None:
         worker.set_key('LOCAL')
+
+    sudo_passwd = task.default("USER_sudo_passwd")
+    if sudo_passwd:
+        worker.write(sudo_passwd.encode() + b'\n')
+
     if task.default("USER_stdin_worker"):
         bind_stdin(worker, display)
+    elif sudo_passwd:
+        # close stdin stream after sudo password forward in that case
```

Comment:
digging further, I think it's a minor bug in clush that is not related to `--sudo`
the stdin stream should always be closed when the `bind_stdin()` is not called.
I will explain in more details in an upcoming separate PR, as we should fix that first.

Reply:
uh? but we don't start the thread in non-interactive mode, so stuff like 'clush -w nodes tar x < foo.tar' won't work anymore if you close stdin when no thread. We should only close if nostdin.

Reply:
Ok I was confused myself. That's my current understanding (90% sure this is correct):

What we know for sure is that stdin is always closed with `--nostdin` because of [this](https://github.com/cea-hpc/clustershell/blob/master/lib/ClusterShell/CLI/Clush.py#L1036) 
This sets a default that force EOF on stdin at [Task.shell()](https://github.com/cea-hpc/clustershell/blob/master/lib/ClusterShell/Task.py#L627). This currently makes the option incompatible with `--sudo`

Now, when `--nostdin` is NOT used... it depends if we are in interactive mode (like `ttyloop()` style interactive) or in non-interactive mode for stuff like 'clush -w nodes tar x < foo.tar' . The difference here is that in interactive mode, we create a new [Task(](https://github.com/cea-hpc/clustershell/blob/master/lib/ClusterShell/CLI/Clush.py#L963)) that creates a new thread for user interaction.

BUT in my understanding, in either case, `bind_stdin()` is called and a dedicated stdin thread is created.

If this stdin thread is NOT created, we don't expect anything from clush's stdin. Still, it might stay open. I believe we should close it if not done already (currently only done already if `--nostdin` is used)

Reply:
Submitted #478 to fix this. Please check. Then I will refresh/rebase this patch on top of it. Thanks!